### PR TITLE
send message in game

### PR DESCRIPTION
**Pull request by tobybrave**
#514

### CODE REVIEW

File: server/src/controllers/game.controller.js [hunk 1]
```diff
@@ -358,10 +356,53 @@ class GameController {
...
+    
+    const messageProps = {
+         text: formattedMessage,
+        // user_name, image_url // user_name & image_url from user info in db
+       };
+    
+    // push to message collection
+    // fetch message collection of game instance, if none create else update
+    if (!gameExist.data.payload.chats) {
+      gameExist.data.payload.chats = [];
```
Comment (eni4sure) [inline server/src/controllers/game.controller.js:388]: the key on the model is ``messages`` not chats

Comment (eni4sure) [reply server/src/controllers/game.controller.js:388]: please fix that everywhere

File: server/src/controllers/game.controller.js [hunk 2]
```diff
@@ -358,10 +356,53 @@ class GameController {
...
+    }
+    gameExist.data.payload.chats = 
+      gameExist.data.payload.chats.concat(messageProps);
+    
+    await GameRepo.update(game_id, gameExist.data.payload);
+    
+    // publish to centrifugo
+    await centrifugoController.publish("chats", messageProps);
```
Comment (eni4sure) [inline server/src/controllers/game.controller.js:396]: where's your payload before publishing to centrifugo?

**Review by eni4sure — CHANGES_REQUESTED** [review]

File: server/src/controllers/game.controller.js [hunk 3]
```diff
@@ -358,10 +356,53 @@ class GameController {
...
+    
+    // players should not be able to send message (to be implemented later)
+    
+    // incase user gets sloppy
+    const formattedMessage = message.trim();
+    if (!formattedMessage) {
+      throw new CustomError("message text cannot be empty", 400);
+    }
+    
+    const messageProps = {
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:380]: The message properties are incomplete…
Please look, at the model

File: server/src/controllers/game.controller.js [hunk 4]
```diff
@@ -358,10 +356,53 @@ class GameController {
...
+         text: formattedMessage,
+        // user_name, image_url // user_name & image_url from user info in db
+       };
+    
+    // push to message collection
+    // fetch message collection of game instance, if none create else update
+    if (!gameExist.data.payload.chats) {
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:387]: Where did ‘payload’ come from?

File: server/src/routes/v1/game.route.js [hunk 5]
```diff
@@ -281,8 +281,8 @@ router.get("/:id", GameCtrl.getById);
 // Get All Games by User - not implemented -- shauib mustapha
 // router.get("/all/:userId", GameCtrl.getAllByUser);
 
-// Send messages to game -- NotImplemented
-// router.patch("/message",GameCtrl.Message);
+// Send messages to game
```
Comment (NdubuisiJr) [inline server/src/routes/v1/game.route.js:284]: Add swagger documentation comment for this route

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Please fix those changes

**Comment by tobybrave** [comment]
all fixed

File: server/src/controllers/game.controller.js [hunk 6]
```diff
@@ -76,7 +76,7 @@ class GameController {
       const payload = {
         event: "join_game",
         permission,
-        player: opponent,
+        player: updated.data.opponent,
```
Comment (eni4sure) [inline server/src/controllers/game.controller.js:79]: switch this to ```opponent``` only, not ``updated.data.opponent``

Comment (tobybrave) [reply server/src/controllers/game.controller.js:79]: I didn't work on that tho
lemme pull dev upstream if it's from the new merge

**Comment by tobybrave** [comment]
done!

File: server/src/controllers/game.controller.js [hunk 7]
```diff
@@ -375,7 +375,63 @@ class GameController {
...
+    // players should not be able to send message (to be implemented later)
+    
+    // incase user gets sloppy
+    const formattedMessage = message.trim();
+    if (!formattedMessage) {
+      throw new CustomError("message text cannot be empty", 400);
+    }
+
+    // test usernames and image_url
+    const testUsernames = ["mark", "jack", "jane", "crystal"];
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:403]: Take these test variables out na…

File: server/src/controllers/game.controller.js [hunk 8]
```diff
@@ -375,7 +375,63 @@ class GameController {
...
+
+    const messageProps = {
+         text: formattedMessage,
+         user_name: testUsernames[random],
+         image_url: testImageUrl // user_name & image_url from user info retrieved from db
+       };
+    
+    // push to message collection
+    // fetch message collection of game instance, if none create else update
+    if (!gameExist.data[0].messages) {
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:415]: Why are you clearing the messages

File: server/src/controllers/game.controller.js [hunk 9]
```diff
@@ -375,7 +375,63 @@ class GameController {
...
+      gameExist.data[0].messages = [];
+    }
+    gameExist.data[0].messages = 
+      gameExist.data[0].messages.concat(messageProps);
+
+    const response = await GameRepo.update(game_id, gameExist.data[0]);
+    
+    console.log(response);
+    // publish to centrifugo
+    await centrifugoController.publish(game_id, {
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:425]: This is not the publishing pattern. Look at other routes publishing to Centrifugo…

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Fix these issues

File: server/src/controllers/game.controller.js (hunk 7 above)
Comment (tobybrave) [reply server/src/controllers/game.controller.js:403]: leave them there for now, once there's a way to fetch the real user info i'd remove and update it

File: server/src/controllers/game.controller.js (hunk 8 above)
Comment (tobybrave) [reply server/src/controllers/game.controller.js:415]: no message is being cleared. it works

File: server/src/controllers/game.controller.js (hunk 9 above)
Comment (tobybrave) [reply server/src/controllers/game.controller.js:425]: what's the publishing pattern then? asides from not assigning it to a payload variable.

File: server/src/controllers/game.controller.js (hunk 6 above)
Comment (eni4sure) [reply server/src/controllers/game.controller.js]: okay boss
fixed that

**Review by eni4sure — APPROVED** [review]

**Comment by eni4sure** [comment]
closing this PR now.
REASON: open for too long

kindly fix the conflicts are open a new PR if the need be

**Pull request closed without merging**